types/lodash/common/collection.d.ts

```diff
@@ -2943,14 +2599,6 @@ declare module "../index" {
             collection: T | null | undefined,
             predicate?: ObjectIterateeCustom<T, boolean>
         ): boolean;
-
-        /**
-         * @see _.some
-         */
-        some<T>(
```

Comment:
For reasons unknown to me, you need to add this overload back to make the tests work in TS 2.4. (same with wrappers)

Reply:
What I found that seems to fix this is not adding back this overload, but changing the definitions of:

```ts
    type ObjectIteratee<TObject> = ObjectIterator<TObject, NotVoid> | string | [string, any] | PartialDeep<TObject[keyof TObject]>;
    type ObjectIterateeCustom<TObject, TResult> = ObjectIterator<TObject, TResult> | string | object | [string, any] | PartialDeep<TObject[keyof TObject]>;
```

to 
```ts
    type ObjectIteratee<TObject> = ObjectIterator<TObject, NotVoid> | string | [string, any] | PartialDeep<TObject>;
    type ObjectIterateeCustom<TObject, TResult> = ObjectIterator<TObject, TResult> | string | object | [string, any] | PartialDeep<TObject>;
```

(focus on the `PartialDeep` part)

I don't understand the purpose of this well enough to know whether this change is correct, or the unit test's usage of `{ a: 42 }` as an iterator is an incorrect unit test, or neither, or both.

It was the `sampleObject` tests of _.some that were failing for me and were fixed by the above change:

```ts
        result = _.some(sampleObject, {a: 42});
```

Reply:
After looking at this some more, I think the test that is failing is actually invalid. It's iterating over properties of `SampleObject`, which are of type `number`, `string` and `boolean`, so the `{a: 42}` iterator doesn't make any sense at all - none of the properties of `SampleObject` are of a type that have a property `a`. 

Do you agree? Should I simply delete the tests that use the `{a: 42}` iterator against a collection of type `SampleObject`?

Reply:
Yeah the test looks invalid. I think PatialDeep needs to stay as it was before (`TObject [keyof TObjet]`) but you should remove the `| object` type (that's my mistake from a while ago). Feel free to remove the test.